### DISCUSSION TITLE
How to use route for shell navigation?

### DISCUSSION THREAD

**Discussion by CodingOctocat** [dotnet/maui#23202] [Q&A]
This is my Shell, there are 4 Tabs, all Tabs have the same button called [Add], when clicked it navigates to AddNewItemPage, AddNewItemPage has a [Confirm] button, when clicked it navigates to ItemPage, note that at this point if you click on the back button, it should go back to the Tab page not to the AddNewItemPage.

I read the docs, but couldn't understand of it：[.NET MAUI Shell navigation](https://learn.microsoft.com/en-us/dotnet/maui/fundamentals/shell/navigation?view=net-maui-8.0)

Is this the right navigation?
```csharp
// RegisterRoute in Shell.xaml.cs
Routes.Add($"{nameof(ItemPage)}", typeof(ItemPage));

// AddNewItemPage.xaml.cs
await Shell.Current.GoToAsync($"../{nameof(ItemPage)}", navigationParameter);
```

```xaml
<?xml version="1.0" encoding="UTF-8" ?>
<Shell x:Class="Demo.Views.AppShell"
       xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
       xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
       x:DataType="{x:Type vm:AppShellViewModel}"
       Shell.FlyoutBehavior="Disabled"
       Shell.ForegroundColor="{StaticResource White}"
       Shell.NavBarHasShadow="True">
    <TabBar>
        <ShellContent Title="Tab1"
                      Route="Tab1" />

        <ShellContent Title="Tab2"
                      Route="Tab2" />

        <ShellContent Title="Tab3"
                      Route="Tab3" />

        <ShellContent Title="Tab4"
                      Route="Tab4" />
    </TabBar>
</Shell>
```

**Comment by prabhav-mehra** [comment]
Bit confused. When you click on the AddNewitemPage it doesn't go the main page (assuming you have a page prior to it) or when you click back on ItemPage it goes to AddNewitemPage?

**Reply by CodingOctocat** [reply]
I tested it. `../ItemPage` is correct.
